src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeModule.cs

```diff
@@ -153,8 +153,6 @@ public override byte[] ResolveSignature(int metadataToken)
 
             Type declaringType = ResolveType(tkDeclaringType, genericTypeArguments, genericMethodArguments);
 
-            declaringType.GetFields();
```

Comment:
Now fields always come in order, no need to load all fields before head

Did not added a test case for this as related tests added with https://github.com/dotnet/runtime/issues/46272